waymo_raw_data_to_argoverse.py

```diff
@@ -30,205 +40,398 @@
+
+
+def undistort_image(img: np.ndarray, calib_data: Any, camera_name: int):
+    """Undistort the image from the Waymo dataset given camera calibration data"""
+    for camera_calib in calib_data:
+        if camera_calib.name == camera_name:
+            f_u, f_v, c_u, c_v, k1, k2, p1, p2, k3 = camera_calib.intrinsic
+            camera_matrix = np.array([[f_u, 0, c_u], [0, f_v, c_v], [0, 0, 1]])
+            dist_coeffs = np.array([k1, k2, p1, p2, k3])
+            return cv2.undistort(img, camera_matrix, dist_coeffs)
+
+
+def form_calibration_json(calib_data):
+    """
+    Argoverse expects to receive "egovehicle_T_camera", i.e. from camera -> egovehicle, with
+            rotation parameterized as quaternion.
+    Waymo provides the same SE(3) transformation, but with rotation parmaeterized as 3x3 matrix
+    """
+    calib_dict = {"camera_data_": []}
+    for camera_calib in calib_data:
+        cam_name = CAMERA_NAMES[camera_calib.name]
+        # They provide "Camera frame to vehicle frame."
+        # https://github.com/waymo-research/waymo-open-dataset/blob/master/waymo_open_dataset/dataset.proto
+        egovehicle_SE3_waymocam = np.array(camera_calib.extrinsic.transform).reshape(
+            4, 4
+        )
+        standardcam_R_waymocam = rotY(-90).dot(rotX(90))
+        standardcam_SE3_waymocam = SE3(
+            rotation=standardcam_R_waymocam, translation=np.zeros(3)
+        )
+        waymocam_SE3_egovehicle = SE3(
+            rotation=egovehicle_SE3_waymocam[:3, :3],
+            translation=egovehicle_SE3_waymocam[:3, 3],
+        )
+        standardcam_SE3_egovehicle = standardcam_SE3_waymocam.right_multiply_with_se3(
+            waymocam_SE3_egovehicle
+        )
+        egovehicle_q_camera = rotmat2quat(standardcam_SE3_egovehicle.rotation)
+        x, y, z = standardcam_SE3_egovehicle.translation
+        qw, qx, qy, qz = egovehicle_q_camera
+        f_u, f_v, c_u, c_v, k1, k2, p1, p2, k3 = camera_calib.intrinsic
+        cam_dict = {
+            "key": "image_raw_" + cam_name,
+            "value": {
+                "focal_length_x_px_": f_u,
+                "focal_length_y_px_": f_v,
+                "focal_center_x_px_": c_u,
+                "focal_center_y_px_": c_v,
+                "skew_": 0,
+                "distortion_coefficients_": [0, 0, 0],
+                "vehicle_SE3_camera_": {
+                    "rotation": {"coefficients": [qw, qx, qy, qz]},
+                    "translation": [x, y, z],
+                },
+            },
+        }
+        calib_dict["camera_data_"] += [cam_dict]
+    return calib_dict
+
+
+def dump_pose(
+    city_SE3_egovehicle: np.ndarray, timestamp: int, log_id: str, parent_path: str
+) -> None:
+    """Saves the SE3 transformation from city frame to egovehicle frame at a particular timestamp
+
+    Args:
+        city_SE3_egovehicle: A (4,4) numpy array representing the SE3 transformation from city to egovehicle frame
```

Comment:
115 char line, can we break down to 80 chars?

Reply:
Addressed [here](https://github.com/johnwlambert/waymo_to_argoverse/pull/2/commits/9db2006d2741dbd66d59c22e07604a9bd62fa2a6)